SU2_CFD/src/output/COutput.cpp

```diff
@@ -424,27 +426,139 @@ void COutput::WriteToFile(CConfig *config, CGeometry *geometry, unsigned short f
           (*fileWritingTable) << "Tecplot ASCII" << fileName + CTecplotFileWriter::fileExt;
       }
 
-      fileWriter = new CTecplotFileWriter(volumeFieldNames, nDim, fileName, volumeDataSorter,
+      fileWriter = new CTecplotFileWriter(fileName, volumeDataSorter,
                                           curTimeIter, GetHistoryFieldValue("TIME_STEP"));
 
       break;
 
+    case PARAVIEW_XML:
+      
+      if (fileName.empty())
+        fileName = config->GetFilename(volumeFilename, "", curTimeIter);
+      
+      /*--- Load and sort the output data and connectivity. ---*/
+      
+      volumeDataSorter->SortConnectivity(config, geometry, true);
+      
+      /*--- Write paraview binary ---*/
+      if (rank == MASTER_NODE) {
+        (*fileWritingTable) << "Paraview" << fileName + CParaviewXMLFileWriter::fileExt;
+      }
+      
+      fileWriter = new CParaviewXMLFileWriter(fileName, volumeDataSorter);
+      
+      break;
+      
     case PARAVIEW_BINARY:
-
+      
       if (fileName.empty())
         fileName = config->GetFilename(volumeFilename, "", curTimeIter);
-
+      
       /*--- Load and sort the output data and connectivity. ---*/
-
+      
       volumeDataSorter->SortConnectivity(config, geometry, true);
-
+      
```

Comment:
Remember to strip all blank spaces :)

Reply:
I wonder why CodeFactor did not complain